src/components/Footer.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useMemo } from 'react';
+import { Filter } from '../types/Filter';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  todos: Todo[];
+  filter: Filter;
+  onFilter: React.Dispatch<React.SetStateAction<Filter>>;
+  clearCompleted: () => void;
+}
+
+export const Footer: React.FC<Props> = ({
+  onFilter,
+  todos,
+  filter,
+  clearCompleted,
+}) => {
+  const counter = useMemo(() => {
+    return todos.filter(todo => !todo.completed).length;
+  }, [todos]);
+
+  const hasCompletedTodos = todos.some(todo => todo.completed);
+
+  const handleFilter = (newFilter: Filter) => (e: React.FormEvent) => {
+    e.preventDefault();
+    onFilter(newFilter);
+  };
+
+  if (!todos.length) {
+    return null;
+  }
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {counter} items left
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={`filter__link ${filter === 'all' ? 'selected' : ''}`}
```

Comment:
consider moving values
'all', 'active', 'completed' to an enum and reusing them where needed